exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -158,109 +171,213 @@ func (e *Exchange) Setup(exch *config.Exchange) error {
+		MessageFilter: connFuturesPrivate,
+		Authenticate:  e.authenticateFuturesAuthConn,
 	})
 }
 
 // FetchTradablePairs returns a list of the exchanges tradable pairs
-func (e *Exchange) FetchTradablePairs(ctx context.Context, _ asset.Item) (currency.Pairs, error) {
-	// TODO: Upgrade to new API version for fetching operational pairs.
-	resp, err := e.GetTicker(ctx)
-	if err != nil {
-		return nil, err
-	}
+func (e *Exchange) FetchTradablePairs(ctx context.Context, assetType asset.Item) (currency.Pairs, error) {
+	switch assetType {
+	case asset.Spot, asset.Margin:
+		resp, err := e.GetExecutionLimits(ctx)
+		if err != nil {
+			return nil, err
+		}
 
-	pairs := make([]currency.Pair, 0, len(resp))
-	for key, info := range resp {
-		// Poloniex returns 0 for highest bid and lowest ask if support has been
-		// dropped from the front end. We don't want to add these pairs.
-		if info.HighestBid == 0 || info.LowestAsk == 0 {
-			continue
+		pairs := make([]currency.Pair, 0, len(resp))
+		for x := range resp {
+			if strings.EqualFold(resp[x].State, "PAUSE") {
+				continue
+			}
+			cp, err := currency.NewPairFromString(resp[x].Symbol)
+			if err != nil {
+				return nil, err
+			}
+			pairs = append(pairs, cp)
 		}
-		var pair currency.Pair
-		pair, err = currency.NewPairFromString(key)
+		return pairs, nil
+	case asset.Futures:
+		instruments, err := e.GetFuturesAllProductInfo(ctx, "")
 		if err != nil {
 			return nil, err
 		}
-		pairs = append(pairs, pair)
+		pairs := make(currency.Pairs, 0, len(instruments))
+		for i := range instruments {
+			if !strings.EqualFold(instruments[i].Status, "Open") {
+				continue
+			}
+			cp, err := currency.NewPairFromString(instruments[i].Symbol)
+			if err != nil {
+				return nil, err
+			}
+			pairs = append(pairs, cp)
+		}
+		return pairs, nil
 	}
-	return pairs, nil
+	return nil, fmt.Errorf("%w %v", asset.ErrNotSupported, assetType)
 }
 
 // UpdateTradablePairs updates the exchanges available pairs and stores
 // them in the exchanges config
 func (e *Exchange) UpdateTradablePairs(ctx context.Context) error {
-	pairs, err := e.FetchTradablePairs(ctx, asset.Spot)
-	if err != nil {
-		return err
-	}
-	if err := e.UpdatePairs(pairs, asset.Spot, false); err != nil {
-		return err
+	enabledAssets := e.GetAssetTypes(true)
+	for _, assetType := range enabledAssets {
+		pairs, err := e.FetchTradablePairs(ctx, assetType)
+		if err != nil {
+			return err
+		}
+		err = e.UpdatePairs(pairs, assetType, false)
+		if err != nil {
+			return err
+		}
 	}
 	return e.EnsureOnePairEnabled()
 }
 
 // UpdateTickers updates the ticker for all currency pairs of a given asset type
-func (e *Exchange) UpdateTickers(ctx context.Context, a asset.Item) error {
-	tick, err := e.GetTicker(ctx)
-	if err != nil {
-		return err
-	}
-
-	enabledPairs, err := e.GetEnabledPairs(a)
+func (e *Exchange) UpdateTickers(ctx context.Context, assetType asset.Item) error {
+	enabledPairs, err := e.GetEnabledPairs(assetType)
```

Comment:
🍺 There is a mismatch between the comment to what this does, it filters to enabled pairs. This has been a slight gotcha when you are trading certain items but are pricing old balances. I think there is not a big need to restrict what they give us, it just adds in bloat, whereas removing this conforms to what the commentary is saying more explicitly.